gemma.cc

```diff
@@ -743,52 +835,37 @@ void ForEachTensor(const Weights<TConfig>* weights,
                    CompressedWeights<TConfig>& c_weights, Func& func) {
   func("c_embedding",
        weights ? weights->embedder_input_embedding.data() : nullptr,
-       c_weights.c_embedder_input_embedding);
+       c_weights.embedder_input_embedding);
   func("c_final_norm", weights ? weights->final_norm_scale.data() : nullptr,
-       c_weights.c_final_norm_scale);
+       c_weights.final_norm_scale);
 
   char name[16];
-  for (int layer_idx = 0; layer_idx < static_cast<int>(TConfig::kLayers);
-       ++layer_idx) {
+  for (int layer_idx = 0; layer_idx < TConfig::kLayers; ++layer_idx) {
     const size_t idx = static_cast<size_t>(layer_idx);
-    Layer<TConfig>* layer = weights ? &weights->layers[idx] : nullptr;
-    CompressedLayer<TConfig>* c_layer = c_weights.CLayer(idx);
-
-    snprintf(name, sizeof(name), "pre_ff_ns_%d", layer_idx);
-    func(name, layer ? layer->pre_ffw_norm_scale.data() : nullptr,
-         c_layer->c_pre_ffw_norm_scale);
-
-    snprintf(name, sizeof(name), "gating_ein_%d", layer_idx);
-    func(name, layer ? layer->gating_einsum_w.data() : nullptr,
-         c_layer->c_gating_einsum_w);
-
-    snprintf(name, sizeof(name), "linear_w_%d", layer_idx);
-    func(name, layer ? layer->linear_w.data() : nullptr, c_layer->c_linear_w);
-    snprintf(name, sizeof(name), "qkv_ein_%d", layer_idx);
-
-    func(name, layer ? layer->qkv_einsum_w.data() : nullptr,
-         c_layer->c_qkv_einsum_w);
-    snprintf(name, sizeof(name), "att_ein_%d", layer_idx);
-
-    func(name, layer ? layer->attn_vec_einsum_w.data() : nullptr,
-         c_layer->c_attn_vec_einsum_w);
-
-    snprintf(name, sizeof(name), "pre_att_ns_%d", layer_idx);
-    func(name, layer ? layer->pre_attention_norm_scale.data() : nullptr,
-         c_layer->c_pre_attention_norm_scale);
+    const Layer<TConfig>* layer = weights ? weights->GetLayer(idx) : nullptr;
+    CompressedLayer<TConfig>* layer_weights = c_weights.GetLayer(idx);
+
+#define CALL_FUNC(memb, n)                          \
```

Comment:
Let's have longer names, e.g. identifier/member and key_str or similar. We could rename `name[16]` to name_buf. Putting the string as the first arg might also make this easier to visually scan, because they are about the same length.